float32[] forces
string[] obj_names
string[] robot_links
float32[] distances